### PR TITLE
UCP/ADDRESS: Use a wrapper structure for holding unpacked address

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6703/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6704/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6705/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6713/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

File: test/gtest/ucp/test_ucp_wireup.cc [hunk 1]
```diff
@@ -432,46 +432,6 @@ UCS_TEST_P(test_ucp_wireup_1sided, multi_wireup) {
     }
 }
 
-UCS_TEST_P(test_ucp_wireup_1sided, reply_ep_send_before) {
```
Comment (evgeny-leksikov) [inline test/gtest/ucp/test_ucp_wireup.cc]: Why these tests are removed?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6726/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov fixed

**Pull request merged**